crates/wash-cli/src/cmd/wit/deps.rs

```diff
@@ -0,0 +1,321 @@
+}
+
+/// Enable extended pull configurations for wkg config
+async fn resolve_extended_pull_configs(
+    pull_cfg: &RegistryPullConfig,
+    wasmcloud_toml_dir: impl AsRef<Path>,
+    wkg_config_overrides: &mut HashMap<String, Override>,
+    wkg_client_config: &mut wasm_pkg_client::Config,
+) -> Result<()> {
+    for RegistryPullSourceOverride { target, source } in pull_cfg.sources.iter() {
+        let (ns, pkgs, _, _, maybe_version) = parse_wit_package_name(target)?;
+        let version_suffix = maybe_version.map(|v| format!("@{v}")).unwrap_or_default();
+
+        match source {
+            // Local files can be used by adding them to the config
+            RegistryPullSource::LocalPath(_) => {
+                let path = source
+                    .resolve_file_path(wasmcloud_toml_dir.as_ref())
+                    .await?;
+                if !tokio::fs::try_exists(&path).await.with_context(|| {
+                    format!(
+                        "failed to check for registry pull source local path [{}]",
+                        path.display()
+                    )
+                })? {
+                    bail!(
+                        "registry pull source path [{}] does not exist",
+                        path.display()
+                    );
+                }
+
+                match (ns, pkgs.as_slice()) {
+                    // namespace-level override
+                    (ns, []) => {
+                        wkg_config_overrides.insert(
+                            ns,
+                            Override {
+                                path: Some(path),
+                                version: None,
+                            },
+                        );
+                    }
+                    // package-level override
+                    (ns, packages) => {
+                        for pkg in packages {
+                            wkg_config_overrides.insert(
+                                format!("{ns}:{pkg}"),
+                                Override {
+                                    path: Some(path.clone()),
+                                    version: None,
+                                },
+                            );
+                        }
+                    }
+                }
+            }
+            RegistryPullSource::Builtin => bail!("no builtins are supported"),
+            RegistryPullSource::RemoteHttp(s) => {
+                let url = Url::parse(s)
+                    .with_context(|| format!("invalid registry pull source url [{s}]"))?;
+                let tempdir = tempfile::tempdir()
+                    .with_context(|| format!("failed to create temp dir for downloading [{url}]"))?
+                    .into_path();
+                let output_path = tempdir.join("unpacked");
+                let http_client = wash_lib::start::get_download_client()?;
+                let req = http_client.get(url.clone()).send().await.with_context(|| {
```

Comment:
Super nit: Maybe we should set an `Accept` header here to tell the server we only acknowledge tarballs?

Reply:
Yeah we have no guarantee the server on the other end will actually honor that, so feels like we might be adding a false sense of safety with that... Another problem is that we don't actually know the server will get it's content type right (it *should*, but it's the internet here...)

Maybe we can add this when someone who has `Accept` parsing as a gate shows up?